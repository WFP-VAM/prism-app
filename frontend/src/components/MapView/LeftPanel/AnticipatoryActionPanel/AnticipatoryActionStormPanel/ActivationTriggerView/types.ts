import { AACategory } from 'context/anticipatoryAction/AAStormStateSlice/parsedStormDataTypes';

export const AAPanelCategories: AACategory[] = [
  AACategory.Severe,
  AACategory.Moderate,
];

export const AADisplayCategory: {
  [key in AACategory]?: string;
} = {
  [AACategory.Severe]: ' >118 KM/H',
  [AACategory.Moderate]: ' >89 KM/H',
};
