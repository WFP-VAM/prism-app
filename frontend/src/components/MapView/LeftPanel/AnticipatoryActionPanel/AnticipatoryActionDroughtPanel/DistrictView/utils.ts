import {
  AnticipatoryActionDataRow,
  AnticipatoryActionState,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { getSeason } from 'context/anticipatoryAction/AADroughtStateSlice/utils';
import { AADataSeverityOrder } from '../utils';
import { TimelineRow } from '../Timeline/utils';

export function districtViewTransform(
  data: AnticipatoryActionDataRow[] | undefined,
  filters: AnticipatoryActionState['filters'],
) {
  if (!data) {
    return undefined;
  }
  const { categories: categoryFilters, selectedIndex, selectedDate } = filters;
  const season = getSeason(selectedDate);

  // Keep valid data only and switch SET phase to 'na' if the READY was not valid
  const validData = data
    .filter(
      x =>
        categoryFilters[x.category] &&
        season === x.season &&
        (selectedIndex === '' || x.index === selectedIndex),
    )
    .map(x => {
      if (x.isValid) {
        return x;
      }
      return {
        ...x,
        phase: 'na',
      } as AnticipatoryActionDataRow;
    });

  const monthsMap = new Map<string, AnticipatoryActionDataRow[]>();
  validData.forEach(x => {
    const val = monthsMap.get(x.date);
    monthsMap.set(x.date, val ? [...val, x] : [x]);
  });

  // eslint-disable-next-line fp/no-mutating-methods
  const months = Array.from(monthsMap.keys()).sort();

  const topFiltered = months
    .map(date => {
      const dateData = monthsMap.get(date);
      if (!dateData) {
        // this should never happen
        throw new Error('Invalid Date');
      }

      if (dateData.every(x => x.computedRow)) {
        return [];
      }

      const setCategories = new Set(
        dateData.filter(x => x.phase === 'Set').map(x => x.category),
      );

      const readyCategories = new Set(
        dateData.filter(x => x.phase === 'Ready').map(x => x.category),
      );

      // only keep data for the highest phase by category
      const ret = dateData.filter(
        x =>
          x.phase === 'Set' ||
          (x.phase === 'Ready' && !setCategories.has(x.category)) ||
          (x.phase === 'na' &&
            !setCategories.has(x.category) &&
            !readyCategories.has(x.category)),
      );

      return ret;
    })
    .flat();

  const sevMap = new Map<number, AnticipatoryActionDataRow[]>();
  topFiltered.forEach(x => {
    // return category data on one line by setting phase bonus to 0
    const sevVal = AADataSeverityOrder(x.category, x.phase, 0);
    const val = sevMap.get(sevVal);
    sevMap.set(sevVal, val ? [...val, x] : [x]);
  });

  const transformed = Object.fromEntries(sevMap);

  return {
    months: Object.fromEntries(
      [...new Set(topFiltered.map(x => x.date))].map(x => [
        x,
        new Date(x).toLocaleString('en-US', { month: 'short' }),
      ]),
    ),
    transformed,
  };
}

export function dateSorter(
  a: [string, AnticipatoryActionDataRow[] | TimelineRow],
  b: [string, AnticipatoryActionDataRow[] | TimelineRow],
) {
  if (a[0] > b[0]) {
    return -1;
  }
  if (a[0] < b[0]) {
    return 1;
  }
  return 0;
}
