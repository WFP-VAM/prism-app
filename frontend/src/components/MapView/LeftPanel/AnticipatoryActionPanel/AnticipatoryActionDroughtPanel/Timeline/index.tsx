import {
  Button,
  Typography,
  createStyles,
  makeStyles,
} from '@material-ui/core';
import {
  AADataSelector,
  AAFiltersSelector,
  AASelectedDistrictSelector,
  setAAFilters,
  setAAView,
} from 'context/anticipatoryAction/AADroughtStateSlice';
import {
  AAView,
  AnticipatoryActionDataRow,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { lightGrey } from 'muiTheme';
import { useDispatch, useSelector } from 'react-redux';
import { useSafeTranslation } from 'i18n';
import { Equalizer, Reply } from '@material-ui/icons';
import { getAAColor, getAAIcon } from '../utils';
import { useAACommonStyles } from '../../utils';
import { dateSorter } from '../DistrictView/utils';
import { timelineTransform } from './utils';

interface TimelineItemProps {
  item: AnticipatoryActionDataRow;
}

function TimelineItem({ item }: TimelineItemProps) {
  const classes = useTimelineItemStyles();
  const { t } = useSafeTranslation();

  const color = getAAColor(item.category, item.isValid ? item.phase : 'na');

  // Calculate font size based on character count
  const calculateFontSize = (text: string) => {
    const baseSize = 0.8; // rem
    const minSize = 0.2; // rem
    const charCount = text.length;
    return Math.max(baseSize - (charCount - 6) * 0.05, minSize);
  };

  const fontSize = calculateFontSize(item.index);

  return (
    <div className={classes.wrapper} style={{ border: `1px solid ${color}` }}>
      <Typography variant="h3">{item.probability}</Typography>
      <Typography>
        {t('trig.')} {item.trigger}
      </Typography>
      <div
        className={classes.probabilityBar}
        style={{
          backgroundColor: color,
          width: `${item.probability * 100}%`,
        }}
      />
      <div
        className={classes.triggerBar}
        style={{
          width: `${item.trigger * 100}%`,
        }}
      />
      <Typography
        className={classes.indexText}
        style={{
          fontSize: `${fontSize}rem`,
        }}
      >
        {item.index}
      </Typography>
    </div>
  );
}

const useTimelineItemStyles = makeStyles(() =>
  createStyles({
    wrapper: {
      width: '100%',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      borderRadius: '2px',
      boxSizing: 'border-box',
      padding: '0.25rem',
    },
    probabilityBar: {
      height: '0.3rem',
      marginBottom: '0.25rem',
      borderRadius: '0 2px 2px 0',
    },
    triggerBar: {
      height: '0.25rem',
      backgroundColor: 'black',
      borderRadius: '0 2px 2px 0',
    },
    indexText: {
      whiteSpace: 'nowrap',
      lineHeight: '1.2rem',
    },
  }),
);

interface TimelineProps {
  dialogs: {
    text: string;
    onclick: () => void;
  }[];
}

function Timeline({ dialogs }: TimelineProps) {
  const classes = useTimelineStyles();
  const commonClasses = useAACommonStyles();
  const { t } = useSafeTranslation();
  const dispatch = useDispatch();
  const AAData = useSelector(AADataSelector);
  const selectedDistrict = useSelector(AASelectedDistrictSelector);
  const AAFilters = useSelector(AAFiltersSelector);

  const timelineButtons = [
    {
      icon: Reply,
      text: 'Back',
      onClick: () => {
        dispatch(setAAFilters({ selectedIndex: '' }));
        dispatch(setAAView(AAView.District));
      },
    },
    {
      icon: Equalizer,
      text: 'Forecast',
      onClick: () => {
        dispatch(setAAFilters({ selectedIndex: '' }));
        dispatch(setAAView(AAView.Forecast));
      },
    },
  ];

  const { windowData, allRows } = timelineTransform({
    filters: AAFilters,
    data: AAData,
    selectedDistrict,
  });

  return (
    <>
      <div className={classes.root}>
        {Object.entries(windowData).map(([win, winData]) => {
          if (!winData || Object.keys(winData.rows).length === 0) {
            return (
              <div key={win} className={classes.windowWrapper}>
                {t('No Data')}{' '}
              </div>
            );
          }
          const { months, rows } = winData;

          return (
            <div key={win} className={classes.windowWrapper}>
              <Typography variant="h4" className={commonClasses.windowHeader}>
                {t(win)}
              </Typography>
              <div className={classes.tableWrapper}>
                <div className={classes.headRowWrapper}>
                  <div className={classes.iconColumn} />
                  {months.map(([date, label]) => (
                    <div key={date} className={classes.headColumn}>
                      <Typography className={classes.monthText}>
                        {t(label)}
                      </Typography>
                    </div>
                  ))}
                </div>
                {
                  // eslint-disable-next-line fp/no-mutating-methods
                  Object.entries({ ...allRows, ...rows })
                    .sort(dateSorter)
                    .map(([rowId, rowData]) => (
                      <div key={rowId} className={classes.rowWrapper}>
                        <div className={classes.iconColumn}>
                          {getAAIcon(
                            rowData.status.category,
                            rowData.status.phase,
                          )}
                        </div>
                        {months.map(([date, _label]) => {
                          const elem = rowData.data.find(z => z.date === date);
                          if (!elem) {
                            return (
                              <div key={date} className={classes.column} />
                            );
                          }
                          return (
                            <div key={date} className={classes.column}>
                              <TimelineItem item={elem} />
                            </div>
                          );
                        })}
                      </div>
                    ))
                }
              </div>
            </div>
          );
        })}
      </div>
      <div className={commonClasses.footerWrapperVert}>
        <div className={commonClasses.footerActionsWrapper}>
          {timelineButtons.map(x => (
            <Button
              key={x.text}
              className={commonClasses.footerButton}
              variant="outlined"
              fullWidth
              onClick={x.onClick}
              startIcon={<x.icon />}
            >
              <Typography>{t(x.text)}</Typography>
            </Button>
          ))}
        </div>
        <div className={commonClasses.footerDialogsWrapperVert}>
          {dialogs.map(dialog => (
            <Typography
              key={dialog.text}
              className={commonClasses.footerDialog}
              component="button"
              onClick={() => dialog.onclick()}
            >
              {t(dialog.text)}
            </Typography>
          ))}
        </div>
      </div>
    </>
  );
}

const useTimelineStyles = makeStyles(() =>
  createStyles({
    root: {
      display: 'flex',
      flexDirection: 'row',
      width: '100%',
      background: lightGrey,
      overflow: 'auto',
      justifyContent: 'space-around',
      overflowY: 'scroll',
      // Browser-specific properties for forcing scrollbar visibility and styling
      '&::-webkit-scrollbar': {
        width: '0.5rem',
        height: '0.5rem',
      },
      '&::-webkit-scrollbar-thumb': {
        background: '#888',
        borderRadius: '0.25rem', // Rounded corners for the scrollbar thumb
      },
      '&::-webkit-scrollbar-thumb:hover': {
        background: '#555',
      },
      '&::-webkit-scrollbar-track': {
        borderRadius: '0.25rem', // Rounded corners for the scrollbar track
      },
    },
    windowWrapper: {
      display: 'flex',
      flexDirection: 'column',
      padding: '0.5rem 0.25rem',
      color: 'black',
    },
    tableWrapper: { display: 'flex', flexDirection: 'column', gap: '2px' },
    headRowWrapper: {
      display: 'flex',
      flexDirection: 'row',
      minHeight: '2.5rem',
      background: 'white',
    },
    rowWrapper: {
      display: 'flex',
      flexDirection: 'row',
      minHeight: '5.3rem',
      background: 'white',
    },
    iconColumn: {
      width: '3rem',
      padding: '0.1rem 0.25rem',
    },
    headColumn: {
      width: '4rem',
      padding: '0.1rem 0.25rem',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    column: {
      width: '4rem',
      padding: '0.1rem 0.25rem',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    monthText: {
      background: lightGrey,
      borderRadius: '4px',
      textAlign: 'center',
      textTransform: 'uppercase',
      lineHeight: '2rem',
      width: '100%',
    },
  }),
);

export default Timeline;
