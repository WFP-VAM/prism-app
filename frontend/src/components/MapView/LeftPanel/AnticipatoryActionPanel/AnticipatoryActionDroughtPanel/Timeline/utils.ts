import { AAWindowKeys } from 'config/utils';
import {
  AACategoryType,
  AAPhase,
  AAPhaseType,
  AAcategory,
  AnticipatoryActionDataRow,
  AnticipatoryActionState,
} from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { getSeason } from 'context/anticipatoryAction/AADroughtStateSlice/utils';

function getColumnKey(val: AnticipatoryActionDataRow): number {
  const { category, phase } = val;
  const catIndex = AAcategory.findIndex(x => x === category);
  const phaseIndex = AAPhase.findIndex(x => x === phase);
  return catIndex * 10 + phaseIndex;
}

export type TimelineRow = {
  status: {
    category: AACategoryType;
    phase: AAPhaseType;
  };
  data: AnticipatoryActionDataRow[];
};

interface TimelineTransformParams {
  filters: AnticipatoryActionState['filters'];
  selectedDistrict: AnticipatoryActionState['selectedDistrict'];
  data: AnticipatoryActionState['data'];
}

export function timelineTransform({
  filters,
  selectedDistrict,
  data,
}: TimelineTransformParams) {
  const { selectedWindow, selectedIndex, categories, selectedDate } = filters;
  const season = getSeason(selectedDate);

  const windowData = (
    selectedWindow === 'All' ? AAWindowKeys : [selectedWindow]
  ).map(win => {
    const districtData = !!selectedDistrict && data[win][selectedDistrict];
    if (!districtData) {
      return [win, null];
    }

    const filtered = districtData.filter(
      x =>
        x.season === season &&
        !x.computedRow &&
        (selectedIndex === '' || selectedIndex === x.index) &&
        categories[x.category],
    );

    const months = [...new Set(filtered.map(x => x.date))].map(x => [
      x,
      new Date(x).toLocaleString('en-US', { month: 'short' }),
    ]);

    const categoriesMap = new Map<
      number,
      {
        status: { category: AACategoryType; phase: AAPhaseType };
        data: AnticipatoryActionDataRow[];
      }
    >();
    filtered.forEach(x => {
      const key = getColumnKey(x);
      const val = categoriesMap.get(key);
      if (val) {
        // Sort the new data based on validity
        // eslint-disable-next-line fp/no-mutating-methods
        const sortedData = [
          x,
          ...val.data.filter(item => item.date === x.date),
        ].sort((a, b) => {
          if (a.isValid && a.isOtherPhaseValid) {
            return -1;
          }
          if (b.isValid && b.isOtherPhaseValid) {
            return 1;
          }
          if (a.isValid) {
            return -1;
          }
          if (b.isValid) {
            return 1;
          }
          return 0;
        });

        // Keep only the highest priority item for this date
        const highestPriorityItem = sortedData[0];

        // Update the data array
        // eslint-disable-next-line fp/no-mutation
        val.data = [
          ...val.data.filter(item => item.date !== x.date),
          highestPriorityItem,
        ];
      } else {
        categoriesMap.set(key, {
          status: {
            category: x.category,
            phase: x.phase,
          },
          data: [x],
        });
      }
    });
    return [win, { months, rows: Object.fromEntries(categoriesMap) }];
  }) as [
    (typeof AAWindowKeys)[number],
    {
      months: string[][];
      rows: { [id: string]: TimelineRow };
    } | null,
  ][];

  const allRows = windowData
    .map(([_win, x]) =>
      Object.entries(x?.rows || {}).map(([id, info]) => [
        id,
        { status: info.status, data: [] },
      ]),
    )
    .flat() as [string, TimelineRow][];

  return {
    windowData: Object.fromEntries(windowData),
    allRows: Object.fromEntries(allRows),
  };
}
