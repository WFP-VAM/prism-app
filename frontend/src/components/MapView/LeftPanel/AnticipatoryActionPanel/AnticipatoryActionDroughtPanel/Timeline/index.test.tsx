import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { AnticipatoryActionState } from 'context/anticipatoryAction/AADroughtStateSlice/types';
import { Panel } from 'config/types';
import { defaultDialogs, mockAAData } from '../test.utils';
import { timelineTransform } from './utils';
import Timeline from '.';

const mockStore = configureStore([]);

const filters: AnticipatoryActionState['filters'] = {
  selectedWindow: 'Window 1',
  categories: {
    Severe: true,
    Moderate: true,
    Normal: true,
    Mild: true,
    na: true,
    ny: true,
  },
  selectedIndex: '',
  selectedDate: '2024-02-01',
};

const store = mockStore({
  mapState: {
    layers: [],
    dateRange: { startDate: 1701432000000 },
    maplibreMap: () => {},
    errors: [],
    layersData: [],
    loadingLayerIds: [],
    boundaryRelationData: {},
  },
  serverState: { availableDates: {}, loading: false },
  anticipatoryActionDroughtState: {
    filters,
    selectedDistrict: 'Changara',
    monitoredDistricts: ['Changara'],
    data: mockAAData,
  },
  leftPanelState: {
    tabValue: Panel.AnticipatoryActionDrought,
  },
});

test('renders as expected', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <Timeline dialogs={defaultDialogs} />
      </Provider>
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();
});

test('District view transformation', () => {
  const res = timelineTransform({
    data: mockAAData,
    filters,
    selectedDistrict: 'Changara',
  });

  expect(res).toEqual(out);
});

const out = {
  windowData: {
    'Window 1': {
      months: [
        ['2023-08-01', 'Aug'],
        ['2023-09-01', 'Sep'],
        ['2023-10-01', 'Oct'],
        ['2023-11-01', 'Nov'],
      ],
      rows: {
        '32': {
          status: {
            category: 'Mild',
            phase: 'Ready',
          },
          data: [
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Ready',
              probability: 0.16,
              trigger: 0.17,
              date: '2023-08-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: false,
            },
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Ready',
              probability: 0.22,
              trigger: 0.3,
              date: '2023-09-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: false,
            },
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Ready',
              probability: 0.31,
              trigger: 0.2,
              date: '2023-10-01',
              season: '2023-24',
              isValid: true,
              isOtherPhaseValid: true,
            },
          ],
        },
        '33': {
          status: {
            category: 'Mild',
            phase: 'Set',
          },
          data: [
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Set',
              probability: 0.12,
              trigger: 0.3,
              date: '2023-09-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: false,
            },
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Set',
              probability: 0.34,
              trigger: 0.33,
              date: '2023-10-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: false,
            },
            {
              category: 'Mild',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: true,
              phase: 'Set',
              probability: 0.28,
              trigger: 0.25,
              date: '2023-11-01',
              season: '2023-24',
              isValid: true,
              isOtherPhaseValid: true,
            },
          ],
        },
        '42': {
          status: {
            category: 'Moderate',
            phase: 'Ready',
          },
          data: [
            {
              category: 'Moderate',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: true,
              phase: 'Ready',
              probability: 0.19,
              trigger: 0.1,
              date: '2023-09-01',
              season: '2023-24',
              isValid: true,
              isOtherPhaseValid: false,
            },
            {
              category: 'Moderate',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Ready',
              probability: 0.28,
              trigger: 0.2,
              date: '2023-10-01',
              season: '2023-24',
              isValid: true,
              isOtherPhaseValid: false,
            },
          ],
        },
        '43': {
          status: {
            category: 'Moderate',
            phase: 'Set',
          },
          data: [
            {
              category: 'Moderate',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Set',
              probability: 0.25,
              trigger: 0.26,
              date: '2023-10-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: true,
            },
            {
              category: 'Moderate',
              district: 'Changara',
              index: 'SPI DJF',
              type: 'SPI',
              window: 'Window 1',
              new: false,
              phase: 'Set',
              probability: 0.22,
              trigger: 0.3,
              date: '2023-11-01',
              season: '2023-24',
              isValid: false,
              isOtherPhaseValid: true,
            },
          ],
        },
      },
    },
  },
  allRows: {
    '32': {
      status: {
        category: 'Mild',
        phase: 'Ready',
      },
      data: [],
    },
    '33': {
      status: {
        category: 'Mild',
        phase: 'Set',
      },
      data: [],
    },
    '42': {
      status: {
        category: 'Moderate',
        phase: 'Ready',
      },
      data: [],
    },
    '43': {
      status: {
        category: 'Moderate',
        phase: 'Set',
      },
      data: [],
    },
  },
};
