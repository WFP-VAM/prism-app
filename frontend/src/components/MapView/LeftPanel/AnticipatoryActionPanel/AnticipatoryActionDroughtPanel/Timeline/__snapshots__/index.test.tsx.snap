// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-windowWrapper-2"
    >
      <mock-typography
        classname="makeStyles-windowHeader-18"
        variant="h4"
      >
        Window 1
      </mock-typography>
      <div
        class="makeStyles-tableWrapper-3"
      >
        <div
          class="makeStyles-headRowWrapper-4"
        >
          <div
            class="makeStyles-iconColumn-6"
          />
          <div
            class="makeStyles-headColumn-7"
          >
            <mock-typography
              classname="makeStyles-monthText-9"
            >
              Aug
            </mock-typography>
          </div>
          <div
            class="makeStyles-headColumn-7"
          >
            <mock-typography
              classname="makeStyles-monthText-9"
            >
              Sep
            </mock-typography>
          </div>
          <div
            class="makeStyles-headColumn-7"
          >
            <mock-typography
              classname="makeStyles-monthText-9"
            >
              Oct
            </mock-typography>
          </div>
          <div
            class="makeStyles-headColumn-7"
          >
            <mock-typography
              classname="makeStyles-monthText-9"
            >
              Nov
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-rowWrapper-5"
        >
          <div
            class="makeStyles-iconColumn-6"
          >
            <div
              class="makeStyles-iconWrapper-19"
              style="background: rgb(255, 137, 52);"
            >
              <div
                class="makeStyles-centerContainer-20"
                style="border: 1px solid black; color: black;"
              >
                <div
                  class="makeStyles-topTextContainer-21"
                  style="border-bottom: 1px solid black; background: rgb(255, 137, 52);"
                >
                  S
                </div>
                <div
                  class="makeStyles-bottomTextContainer-22"
                  style="background: rgb(255, 137, 52);"
                >
                  MOD
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          />
          <div
            class="makeStyles-column-8"
          />
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.25
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.26
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 25%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 26%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.22
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.3
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 22%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 30%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-rowWrapper-5"
        >
          <div
            class="makeStyles-iconColumn-6"
          >
            <div
              class="makeStyles-iconWrapper-19"
              style="background: rgb(255, 213, 45);"
            >
              <div
                class="makeStyles-centerContainer-20"
                style="border: 1px solid black; color: black;"
              >
                <div
                  class="makeStyles-topTextContainer-21"
                  style="border-bottom: 1px solid black; background: rgb(255, 213, 45);"
                >
                  R
                </div>
                <div
                  class="makeStyles-bottomTextContainer-22"
                  style="background: rgb(255, 213, 45);"
                >
                  MOD
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          />
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #ffd52d;"
            >
              <mock-typography
                variant="h3"
              >
                0.19
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.1
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(255, 213, 45); width: 19%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 10%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #ffd52d;"
            >
              <mock-typography
                variant="h3"
              >
                0.28
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.2
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(255, 213, 45); width: 28.000000000000004%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 20%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          />
        </div>
        <div
          class="makeStyles-rowWrapper-5"
        >
          <div
            class="makeStyles-iconColumn-6"
          >
            <div
              class="makeStyles-iconWrapper-19"
              style="background: rgb(255, 245, 3);"
            >
              <div
                class="makeStyles-centerContainer-20"
                style="border: 1px solid black; color: black;"
              >
                <div
                  class="makeStyles-topTextContainer-21"
                  style="border-bottom: 1px solid black; background: rgb(255, 245, 3);"
                >
                  S
                </div>
                <div
                  class="makeStyles-bottomTextContainer-22"
                  style="background: rgb(255, 245, 3);"
                >
                  MIL
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          />
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.12
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.3
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 12%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 30%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.34
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.33
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 34%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 33%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #fff503;"
            >
              <mock-typography
                variant="h3"
              >
                0.28
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.25
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(255, 245, 3); width: 28.000000000000004%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 25%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-rowWrapper-5"
        >
          <div
            class="makeStyles-iconColumn-6"
          >
            <div
              class="makeStyles-iconWrapper-19"
              style="background: rgb(255, 252, 179);"
            >
              <div
                class="makeStyles-centerContainer-20"
                style="border: 1px solid black; color: black;"
              >
                <div
                  class="makeStyles-topTextContainer-21"
                  style="border-bottom: 1px solid black; background: rgb(255, 252, 179);"
                >
                  R
                </div>
                <div
                  class="makeStyles-bottomTextContainer-22"
                  style="background: rgb(255, 252, 179);"
                >
                  MIL
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.16
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.17
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 16%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 17%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #f1f1f1;"
            >
              <mock-typography
                variant="h3"
              >
                0.22
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.3
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(241, 241, 241); width: 22%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 30%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          >
            <div
              class="makeStyles-wrapper-23"
              style="border: 1px solid #fffcb3;"
            >
              <mock-typography
                variant="h3"
              >
                0.31
              </mock-typography>
              <mock-typography>
                trig.
                 
                0.2
              </mock-typography>
              <div
                class="makeStyles-probabilityBar-24"
                style="background-color: rgb(255, 252, 179); width: 31%;"
              />
              <div
                class="makeStyles-triggerBar-25"
                style="width: 20%;"
              />
              <mock-typography
                classname="makeStyles-indexText-26"
                style="font-size: 0.75rem;"
              >
                SPI DJF
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-column-8"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-footerWrapperVert-15"
  >
    <div
      class="makeStyles-footerActionsWrapper-11"
    >
      <mock-button
        classname="makeStyles-footerButton-13"
        fullwidth="true"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Back
        </mock-typography>
      </mock-button>
      <mock-button
        classname="makeStyles-footerButton-13"
        fullwidth="true"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Forecast
        </mock-typography>
      </mock-button>
    </div>
    <div
      class="makeStyles-footerDialogsWrapperVert-16"
    >
      <mock-typography
        classname="makeStyles-footerDialog-14"
        component="button"
      >
        How to read this screen
      </mock-typography>
    </div>
  </div>
</div>
`;
